ozaria/site/views/play/level/PlayLevelView.js

```diff
@@ -1394,13 +1419,23 @@ class PlayLevelView extends RootView {
     if (this.destroyed) {
       return
     }
+
     Backbone.Mediator.publish('level:set-time', { ratio: 1 })
+
+    // Don't award achievements when beating level changed in level editor
     if (this.level.hasLocalChanges()) {
       return
-    } // Don't award achievements when beating level changed in level editor
-    if (this.goalManager.checkOverallStatus() === 'success') {
+    }
+
+    const options = { showModal: true }
+    const state = this.session.get('state')
+
+    if (this.goalManager.finishLevel()) {
+      if (state && state.capstoneStage && this.goalManager.getRemainingGoals().length === 0) {
```

Comment:
Does this mean the capstone is still in progress or fully complete? Where do we publish show-victory event without `capstoneInProgress` such that overall level session state is set as `complete: true`

Reply:
It means that the capstone is still in progress. When there is no `capstoneInProgress`, it is done and goes through the regular path: setting `complete: true`. 

Or said another way, as long as `capstoneInProgress` is passed along in the options, the level is not fully complete.

Reply:
I am not sure if I understand why the length of the remaining goals is 0 but the level is still in progress?

Reply:
You're completely right - I changed this on the other machine and missed pushing it up. Updated now.